cwlVersion: v1.0
class: Workflow


requirements:
  - class: StepInputExpressionRequirement
  - class: InlineJavascriptRequirement


inputs:

  annotation_file:
    type: File
    label: "TSV annotation file"
    format: "http://edamontology.org/format_3475"
    doc: "TSV annotation file"

  bambai_pair:
    type: File
    secondaryFiles:
    - .bai
    label: "Coordinate sorted BAM+BAI files"
    format: "http://edamontology.org/format_2572"
    doc: "Coordinate sorted BAM file and BAI index file"

  isoforms_file:
    type: File
    label: "CSV isoforms file"
    format: "http://edamontology.org/format_3752"
    doc: "CSV isoforms file generated by GEEP"

  stats_file:
    type: File
    label: "Stat file"
    format: "http://edamontology.org/format_2330"
    doc: "Stat file generated by python-get-stat-rnaseq.cwl"

  output_prefix:
    type: string?
    label: "Output file prefix"
    doc: "Output file prefix"

  pair:
    type: boolean?
    label: "Paired end experiment type"
    doc: "TRUE if paired end experiment type"

  dutp:
    type: boolean?
    label: "dUTP experiment type"
    doc: "TRUE if dUTP experiment type"

  threads:
    type: int?
    label: "Number of threads"
    doc: "Number of threads"


outputs:

  png_file:
    type: File[]
    label: "PNG plots"
    format: "http://edamontology.org/format_3603"
    doc: "R generated coverage and RPKM histogram plots"
    outputSource: plot_rna/png_file

  coverage_file:
    type: File
    label: "Coverage"
    format: "http://edamontology.org/format_3475"
    doc: "Coverage plot data"
    outputSource: plot_rna/coverage_file

  rpkm_file:
    type: File
    label: "RPKMs"
    format: "http://edamontology.org/format_3475"
    doc: "RPKM histogram data"
    outputSource: plot_rna/rpkm_file


steps:

  plot_rna:
    in:
      annotation_file: annotation_file
      bambai_pair: bambai_pair
      isoforms_file: isoforms_file
      stats_file: stats_file
      output_prefix: output_prefix
      pair: pair
      dutp: dutp
      threads: threads
    out: [png_file, coverage_file, rpkm_file]
    run:
      cwlVersion: v1.0
      class: CommandLineTool
      requirements:
      - class: InlineJavascriptRequirement
        expressionLib:
        - var get_output_prefix = function(ext) {
              ext = ext || "";
              if (inputs.output_prefix == ""){
                let root = inputs.bambai_pair.basename.split('.').slice(0,-1).join('.');
                return (root == "")?inputs.bambai_pair.basename+ext:root+ext;
              } else {
                return inputs.output_prefix;
              }
          }
      - class: DockerRequirement
        dockerPull: biowardrobe2/plugin-plot-rna:v0.0.2
      inputs:
        annotation_file:
          type: File
          inputBinding:
            position: 5
            prefix: "-a"
          doc: "TSV annotation file"
        bambai_pair:
          type: File
          inputBinding:
            position: 6
            prefix: "-b"
          secondaryFiles:
            - .bai
          doc: "Indexed BAM + BAI files"
        isoforms_file:
          type: File
          inputBinding:
            position: 7
            prefix: "-i"
          doc: "CSV isoforms file generated by GEEP"
        stats_file:
          type: File
          inputBinding:
            position: 8
            prefix: "-s"
          doc: "Stat file generated by python-get-stat-rnaseq.cwl"
        output_prefix:
          type: string?
          inputBinding:
            position: 9
            prefix: "-o"
            valueFrom: $(get_output_prefix("_default_"))
          default: ""
          doc: "Output file prefix"
        pair:
          type: boolean?
          inputBinding:
            position: 10
            prefix: "-p"
          doc: "TRUE if paired end experiment type"
        dutp:
          type: boolean?
          inputBinding:
            position: 11
            prefix: "-d"
          doc: "TRUE if dUTP experiment type"
        threads:
          type: int?
          inputBinding:
            position: 12
            prefix: "-t"
          doc: "Number of threads"
      outputs:
        png_file:
          type: File[]
          outputBinding:
            glob: "*.png"
        coverage_file:
          type: File
          outputBinding:
            glob: "*cov.tsv"
        rpkm_file:
          type: File
          outputBinding:
            glob: "*rpkm.tsv"
      baseCommand: ["plot_rna.R"]


$namespaces:
  s: http://schema.org/

$schemas:
- http://schema.org/docs/schema_org_rdfa.html

s:name: "plot-rna"
s:downloadUrl: https://raw.githubusercontent.com/michael-kotliar/biowardrobe-airflow-plugins/cwls/plot-rna.cwl
s:codeRepository: https://github.com/michael-kotliar/biowardrobe-airflow-plugins
s:license: http://www.apache.org/licenses/LICENSE-2.0

s:isPartOf:
  class: s:CreativeWork
  s:name: Common Workflow Language
  s:url: http://commonwl.org/

s:creator:
- class: s:Organization
  s:legalName: "Cincinnati Children's Hospital Medical Center"
  s:location:
  - class: s:PostalAddress
    s:addressCountry: "USA"
    s:addressLocality: "Cincinnati"
    s:addressRegion: "OH"
    s:postalCode: "45229"
    s:streetAddress: "3333 Burnet Ave"
    s:telephone: "+1(513)636-4200"
  s:logo: "https://www.cincinnatichildrens.org/-/media/cincinnati%20childrens/global%20shared/childrens-logo-new.png"
  s:department:
  - class: s:Organization
    s:legalName: "Allergy and Immunology"
    s:department:
    - class: s:Organization
      s:legalName: "Barski Research Lab"
      s:member:
      - class: s:Person
        s:name: Michael Kotliar
        s:email: mailto:misha.kotliar@gmail.com
        s:sameAs:
        - id: http://orcid.org/0000-0002-6486-3898

doc: |
  Workflow to plot RNA-Seq results

s:about: |
  Workflow to plot RNA-Seq results
